feat: añadir comprobaciones al crear cliente

JuanAntonioMontes opened the pull request:
Se han añadido comprobaciones de que existan tanto el país como la dirección, y se ha modificado el GlobalHandler para capturar el error y el test unitario CreateClientTest.

JuanAntonioMontes committed:
feat: añadir comprobaciones al crear cliente

Se han añadido comprobaciones de que existan tanto el país como la dirección, y se ha modificado el GlobalHandler para capturar el error y el test unitario CreateClientTest.

RoiGz commented on src/main/java/com/helloworld/renting/service/client/ClientService.java:
```diff
@@ -64,12 +64,36 @@ public ClientDto createClient(ClientDto clientDto) {
             throw new InvalidClientDtoException("El scoring no puede ser nulo ni negativo");
         }
 
+        // Validación de existencia del país
```
Yo, como mínimo, pasaría estas validaciones a un método privado que lance una excepción si no se valida algo

JuanAntonioMontes replied:
Estoy refactorizándolo, ahora hago push de la rama de nuevo

RoiGz commented on src/test/java/com/helloworld/renting/service/client/CreateClientTest.java:
Me gustan bastante los tests

RoiGz reviewed (changes requested):
Tan solo el tema de clean code de tamaño del método que os he puesto

JuanAntonioMontes committed:
feat: refactorizar

-Refactorización de las validaciones
-Se ha capturado el error de telefono duplicado tras ver que devolvia un error 500 de BBDD
